pandas/core/computation/expr.py

```diff
@@ -327,7 +328,7 @@ class BaseExprVisitor(ast.NodeVisitor):
     parser : str
     preparser : callable
     """
-    const_type = Constant
+    const_type = Constant  # type: Type[Term]
```

Comment:
I'm a little unsatisfied with this approach. Both pandas.core.computation.ops and pandas.core.computation.pytables define a `Constant` class inheriting from pandas.core.computation.ops.Term. 

Here in expr.py, the `BaseExprVisitor` class assigns the ops `Constant` to its `const_type` variable. Pytables.py defines an `ExprVisitor` class, which inherits from `BaseExprVisitor` and then assigns its own version of `Constant` to `const_type`.

This throws the mypy error: `pandas/core/computation/pytables.py:384: error: Incompatible types in assignment (expression has type "Type[pandas.core.computation.pytables.Constant]", base class "BaseExprVisitor" defined the type as "Type[pandas.core.computation.ops.Constant]")`.

This approach works because we're saying "`const_type` is always a subclass of `Term`." As far as I can tell, nothing else inherits from `Term`, so there's no problem with that.

Another solution would be to import the pandas.core.computation.pytables version of Constant as Pytables_Constant and replace this type hint with `# type: Type[Union[Constant, Pytables_Constant]]`. But that feels a bit verbose and inelegant when this approach works. I also don't love adding a somewhat confusing import solely for that purpose.